### PR TITLE
[EPMEDU-797]: Auto-cancel feeding booking once the feeding timer expired

### PR DESCRIPTION
**Pull request by Andrii-Chornyi**
- Created dialog which appears on the timer expiration. The user is forcefully redirected to the Home screen from any other screen when the timer expires.
- Added Yellow coloring for reserved feeding points (partially covers EPMEDU-791) to demonstrate the change of state.

<details>
  <summary> Demo: </summary>

https://user-images.githubusercontent.com/112079677/218881767-07c0263d-5422-4399-8220-5df0fec741b3.mp4

</details>

### CODE REVIEW

File: feature/home/src/main/java/com/epmedu/animeal/home/presentation/ui/map/RouteView.kt [hunk 1]
```diff
@@ -40,6 +40,9 @@ internal fun RouteView(
             FeedingRouteState.Disabled -> {
                 mapView.removeRoute(mapBoxRouteInitOptions)
             }
+            FeedingRouteState.TimerExpired -> {
```
Comment (Fileek) [inline feature/home/src/main/java/com/epmedu/animeal/home/presentation/ui/map/RouteView.kt]: You can connect these 2 blocks using `,`: `FeedingRouteState.Disabled, FeedingRouteState.TimerExpired -> ...`

File: feature/home/src/main/java/com/epmedu/animeal/home/presentation/ui/FeedingExpiredDialog.kt [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import com.epmedu.animeal.foundation.preview.AnimealPreview
+import com.epmedu.animeal.foundation.theme.AnimealTheme
+import com.epmedu.animeal.resources.R
+
+@Composable
+internal fun FeedingExpiredDialog(
+    onConfirm: () -> Unit,
+    content: @Composable (() -> Unit)? = null,
+) {
+    AlertDialog(
```
Comment (Fileek) [inline feature/home/src/main/java/com/epmedu/animeal/home/presentation/ui/FeedingExpiredDialog.kt]: Can you move this dialog to `library` module, name it `AnimealAlertDialog` and rename current `AnimealAlertDialog` to `AnimealQuestionDialog`?

File: feature/home/src/main/java/com/epmedu/animeal/home/presentation/ui/FeedingExpiredDialog.kt [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+                fontSize = 18.sp,
+                style = MaterialTheme.typography.h6
+            )
+        },
+        text = content,
+        buttons = {
+            AnimealButton(
+                modifier = Modifier.fillMaxWidth().padding(24.dp),
+                text = stringResource(id = R.string.got_it),
+                onClick = { onConfirm() }
```
Comment (Fileek) [inline feature/home/src/main/java/com/epmedu/animeal/home/presentation/ui/FeedingExpiredDialog.kt]: There is no need to create a lambda, just pass the callback: `onClick = onConfirm`

**Review by Fileek — CHANGES_REQUESTED** [review]
Please refactor `onTimerExpire` lambda to using UI state + events approach

**Comment by Andrii-Chornyi** [comment]
> Please refactor `onTimerExpire` lambda to using UI state + events approach

Moved all timer logic to tabs module

File: feature/home/src/main/java/com/epmedu/animeal/home/presentation/HomeScreenUI.kt [hunk 4]
```diff
@@ -37,25 +37,36 @@ import kotlinx.coroutines.launch
...
-                onScreenEvent(HomeScreenEvent.RouteEvent.FeedingRouteCancellationRequest)
-            }
-        )
+    val hideBottomSheet: () -> Unit = {
+        scope.launch {
+            bottomSheetState.hide()
+        }
     }
 
-    scope.launch {
```
Comment (Andrii-Chornyi) [inline feature/home/src/main/java/com/epmedu/animeal/home/presentation/HomeScreenUI.kt]: I'm uncertain that this condition was needed, as it completely blocks viewing bottom sheet during showing route to feeding points, according to my ticket to Figma designs accessing bottom sheet should be possible. In ticket EPMEDU-883, which added this code, there are no such requirements as well. Please, correct me if I'm wrong and it should remain

File: feature/home/src/main/java/com/epmedu/animeal/home/presentation/viewmodel/handlers/feedingpoint/FeedingPointHandler.kt [hunk 5]
```diff
@@ -8,7 +8,7 @@ internal interface FeedingPointHandler {
 
     suspend fun fetchFeedingPoints()
 
-    fun hideOtherFeedingPoints(feedingPoint: FeedingPointModel)
```
Comment (Andrii-Chornyi) [inline feature/home/src/main/java/com/epmedu/animeal/home/presentation/viewmodel/handlers/feedingpoint/FeedingPointHandler.kt:11]: renamed this function as it got additional functionality of applying visual "reserved" look for feeding point

File: shared/feature/timer/src/main/java/com/epmedu/animeal/timer/data/repository/TimerRepository.kt [hunk 6]
```diff
@@ -0,0 +1,13 @@
+package com.epmedu.animeal.timer.data.repository
```
Comment (Fileek) [inline shared/feature/timer/src/main/java/com/epmedu/animeal/timer/data/repository/TimerRepository.kt]: Please move the repository interface to the domain layer (create usecase package for usecases)

**Review by Fileek — COMMENTED** [review]
Please try to find a better solution to use a timer with flow

**Review by Fileek — APPROVED** [review]

**Pull request merged**